src/components/Form.tsx

```diff
@@ -76,34 +56,49 @@ export const Form = ({ showForm, setShowForm }: Props): JSX.Element => {
     function cancel(): void {
       if (selectedOption && !showCancelConfirmation) {
         setShowCancelConfirmation(true);
-        setCancelText('Confirm?');
         return;
       }
       closeForm();
     }
 
     function closeForm(): void {
       setShowForm(false);
+      setTimeout(() => {
+        setSelectedOption(null);
+        setShowCancelConfirmation(false);
+      }, 500);
     }
 
     return (
       <div className="flex flex-row-reverse gap-2 mt-4">
         <button
           type="button"
           className="px-6 py-2 enabled:bg-custom-teal enabled:hover:bg-teal-700 disabled:bg-gray-400 text-white rounded"
-          onClick={() => save()}
-          disabled={disableSave}
+          onClick={save}
+          disabled={!selectedOption}
         >
           Save
         </button>
 
-        <button
-          type="button"
-          className="px-4 py-2 bg-red-500 hover:bg-red-700 text-white rounded"
-          onClick={() => cancel()}
-        >
-          {cancelText}
-        </button>
+        {!showCancelConfirmation && (
+          <button
+            type="button"
+            className="px-4 py-2 bg-red-500 hover:bg-red-700 text-white rounded"
+            onClick={cancel}
+          >
+            Cancel
+          </button>
+        )}
+
+        {showCancelConfirmation && (
+          <button
+            type="button"
+            className="px-4 py-2 bg-red-500 hover:bg-red-700 text-white rounded"
+            onClick={cancel}
```

Comment:
why not like this to less complicate?
```
onClick={closeForm}
```

Reply:
updated in https://github.com/RR-Ardhika/murojaah/pull/6/commits/b2b332f493123bb553bfb39bdbb103f049bf7ae4, thank you